### DISCUSSION TITLE
Failing to run due to a "wireguard module not active" message on a Synology NAS.

### DISCUSSION THREAD

**Discussion by Subject2Risk** [bubuntux/nordlynx#139] [General]
TL;DR  `bubuntux/nordvpn` was working at one point, now it doesn't so I'm trying to get `bubuntux/nordlynx` going and it fails with a "wireguard module not active" message.

I had some success a couple years back get the `bubuntux/nordvpn` image running, but only up to version 3.9.1 (I think).   I did it using a `docker run` command line with `USER/PASS` options on the command line and adding the all important `/dev/net/tun` device and `tun.ko` module on the NAS.  

Fast forward to today (all day!) and I've been trying to get this version of a `docker_compose.yml` file going but it fails:

    version: "3"
    services:
      nordlynx:
        container_name: nordlynx
        image: ghcr.io/bubuntux/nordlynx
        restart: unless-stopped
        cap_add:
          - NET_ADMIN #required
          - NET_RAW
        environment:
          - PRIVATE_KEY{My Private Key}
          - NET_LOCAL=192.168.1.20/24
        ports:
          - 8080:8080
          - 9090:9090
        sysctls:
          - net.ipv6.conf.all.disable_ipv6=1
        devices:
          - /dev/net/tun

When I run the above `docker-compose.yml` file I get this:

    $ sudo docker-compose up
    Password: 
    [+] Running 1/0
     ⠿ Container nordlynx  Created                                                                                               0.0s
    Attaching to nordlynx
    nordlynx  | [migrations] started
    nordlynx  | [migrations] no migrations found
    nordlynx  | usermod: no changes
    nordlynx  | [2023-07-03T20:41:32+00:00] Firewall is up, everything has to go through the vpn
    nordlynx  | ───────────────────────────────────────
    nordlynx  |   _____ __ __ _____ _____ _____ _____ 
    nordlynx  |  |     |  |  |   __|_   _|     |     |
    nordlynx  |  |   --|  |  |__   | | | |  |  | | | |
    nordlynx  |  |_____|_____|_____| |_| |_____|_|_|_|
    nordlynx  |        _____ __ __ _ __    ____  
    nordlynx  |       | __  |  |  | |  |  |    \ 
    nordlynx  |       | __ -|  |  | |  |__|  |  |
    nordlynx  |       |_____|_____|_|_____|____/ 
    nordlynx  | 
    nordlynx  |   Based on images from linuxserver.io
    nordlynx  | ───────────────────────────────────────
    nordlynx  | 
    nordlynx  | To support the app dev(s) visit:
    nordlynx  | Bubuntux: https://github.com/sponsors/bubuntux
    nordlynx  | WireGuard: https://www.wireguard.com/donations/
    nordlynx  | To support LSIO projects visit:
    nordlynx  | https://www.linuxserver.io/donate/
    nordlynx  | 
    nordlynx  | ───────────────────────────────────────
    nordlynx  | GID/UID
    nordlynx  | ───────────────────────────────────────
    nordlynx  | 
    nordlynx  | User UID:    911
    nordlynx  | User GID:    911
    nordlynx  | ───────────────────────────────────────
    nordlynx  | 
    nordlynx  | RTNETLINK answers: Not supported
    nordlynx  | [2023-07-03T20:41:32+00:00] The wireguard module is not active, try `docker run --rm --cap-add=NET_ADMIN --cap-add=SYS_MODULE -v /lib/modules:/lib/modules bubuntux/install-wireguard` to install it or follow the proper instructions from https://www.wireguard.com/install/ to manually install it.
    ^CGracefully stopping... (press Ctrl+C again to force)
    Aborting on container exit...

The concerning part is that I'm being told that the "wiregaurd module is not active": _is it not active in the image, or on the host_?  (Tried running the docker command suggested but it failed on the NAS because the install process couldn't find kernel headers - not sure where I'd get those for the Synology NAS :/ ).

**Comment by jca2112** [comment]
Just wanted to add I had the same issue also come up 2 days ago. I'm not sure if it can be associated with a recent Synology DSM update or not. (I believe Synology changed their Docker app, now called "Container" -- I am using Portainer, not the Synology Docker app.)

I've been running nordlynx without issue for a while now, until the previously mentioned error:

_The wireguard module is not active, try `docker run --rm --cap-add=NET_ADMIN --cap-add=SYS_MODULE -v /lib/modules:/lib/modules bubuntux/install-wireguard` to install it or follow the proper instructions from https://www.wireguard.com/install/ to manually install it._

Any ideas how on how to proceed? (I didn't try the suggested docker run command that the original poster said did not work -- also wondering if Wireguard is supposed to be installed in the docker image or on the Synology itself?)

**Reply by jca2112** [reply]
Ok, I looked into this a bit further. It looks like a Synology 7.2 DSM update broke the previously (compiled) Wireguard SPK package running on the Synology. (It will no longer run.) I followed the instructions on this page:

https://www.blackvoid.club/wireguard-spk-for-your-synology-nas/

And downloaded a new Wireguard package updated for DSM 7.2. Uninstalled the previous Wireguard package in the Package Center. Installed the new Wireguard package (without running after install), rebooted the Synology, ran the mentioned start script via SSH, which got Wireguard running again. I then had to re-deploy all the Docker images that used the previous nordlynx container (had to edit all the previous images to use the nordlynx container as the network again), which seemed to get everything back up and running. (I have not tested for any VPN leaks, issues, etc. yet though.)
